mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -635,6 +636,15 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
       }
     }
 
+    private def copyToTri(): Unit = {
+      var ii = 0
+      for(i <- 0 until k)
```

Comment:
I think this might fail the style check for missing space before the paren.
Also, I think the received wisdom is that for loops are slow in Scala? if this is performance-critical, you may convert to while loops. Also you can cache i*k from the outer loop below in the inner loop